src/webapp/pages/form-page/useForm.ts

```diff
@@ -226,6 +234,84 @@ export function useForm(formType: FormType, id?: Id): State {
         }
     }, [configurableForm, configurations, formState.kind, isIncidentManager]);
 
+    const handleRemove = useCallback(
+        (id: string) => {
+            if (formState.kind !== "loaded" || !entityData) return;
+
+            switch (entityData.type) {
+                case "risk-assessment-questionnaire": {
+                    const sectionIndexToDelete = formState.data.sections
+                        .filter(section => section.title?.includes("Custom Question"))
+                        .findIndex(section => section.id === id);
+
+                    const entityId = entityData.entity?.additionalQuestions?.find(
+                        (_, index) => index === sectionIndexToDelete
+                    )?.id;
+
+                    if (entityId) {
+                        setFormSectionsToDelete(prevState => [...prevState, entityId]);
+
+                        const updatedEntityData: ConfigurableForm = {
+                            ...entityData,
+                            entity: {
+                                ...entityData.entity,
+                                additionalQuestions: entityData.entity?.additionalQuestions?.filter(
+                                    (_, index) => index !== sectionIndexToDelete
+                                ),
+                            } as RiskAssessmentQuestionnaire,
+                        };
+
+                        setEntityData(updatedEntityData);
+                    }
+                    break;
+                }
+                case "incident-response-actions": {
+                    const sectionIndexToDelete = formState.data.sections.findIndex(
+                        section => section.id === id
+                    );
+
+                    const entityId = entityData.entity?.find(
+                        (_, index) => index === sectionIndexToDelete
+                    )?.id;
+
+                    if (entityId) {
+                        setFormSectionsToDelete(prevState => [...prevState, entityId]);
+
+                        const updatedEntityData: ConfigurableForm = {
+                            ...entityData,
+                            entity: entityData.entity.filter(
+                                (_, index) => index !== sectionIndexToDelete
+                            ),
+                        };
+                        setEntityData(updatedEntityData);
+                    }
+                    break;
+                }
+                default:
+                    break;
+            }
+
+            setFormState(prevState => {
+                if (prevState.kind === "loaded") {
+                    const formSections = prevState.data.sections.filter(
+                        section => section.id !== id
+                    );
+
+                    return {
```

Comment:
Here we should check if form is valid with the new sections to disable or not the save button

[Screencast from 2024-11-28 12-03-12.webm](https://github.com/user-attachments/assets/10017013-6332-484f-b9c8-a3c5da3e23ad)